x/auctions/keeper/msg_server_test.go

```diff
@@ -0,0 +1,387 @@
+		},
+		{
+			name: "fail when sender is not valid bech32",
+			malleate: func() {
+			},
+			input: func() *types.MsgBid {
+				return &types.MsgBid{
+					Sender: "",
+					Amount: sdk.NewCoin(utils.BaseDenom, bidAmount),
+				}
+			},
+			postCheck:   func() {},
+			expErr:      true,
+			errContains: "invalid sender address",
+		},
+		{
+			name: "fail when sender does not have enough funds",
+			malleate: func() {
+			},
+			input: func() *types.MsgBid {
+				return &types.MsgBid{
+					Sender: emptyAddress.String(),
+					Amount: sdk.NewCoin(utils.BaseDenom, bidAmount),
+				}
+			},
+			postCheck:   func() {},
+			expErr:      true,
+			errContains: "transfer bid coins failed",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			keyring := testkeyring.New(1)
+			network = testnetwork.NewUnitTestNetwork(
+				testnetwork.WithPreFundedAccounts(keyring.GetAllAccAddrs()...),
+			)
+
+			validSenderKey = keyring.GetKey(0)
+
+			// Update chain environment before executing the message.
+			tc.malleate()
+			_, err := network.App.AuctionsKeeper.Bid(network.GetContext(), tc.input())
+
+			if tc.expErr {
+				assert.Error(t, err, "expected error but got nil")
+				assert.Contains(t, err.Error(), tc.errContains, "expected different error for test case: %s", tc.name)
+			} else {
+				assert.NoError(t, err, "error not expected")
+			}
+
+			tc.postCheck()
+		})
+	}
+}
+
+func TestDepositCoin(t *testing.T) {
+	var network *testnetwork.UnitTestNetwork
+
+	// validSenderKey is populated during each
+	// run and used to create the MsgDepositCoin.
+	var validSenderKey testkeyring.Key
+	depositAmount := sdk.NewInt(5)
+	emptyAddress, _ := testutiltx.NewAccAddressAndKey()
+
+	testCases := []struct {
+		name        string
+		malleate    func()                       // used to modify the initial conditions of the test or for pre test checks
+		input       func() *types.MsgDepositCoin // return the message used for the bid
+		postCheck   func()
+		expErr      bool
+		errContains string
+	}{
+		{
+			name: "pass",
+			malleate: func() {
+				auctionCollectorAddress := network.App.AccountKeeper.GetModuleAddress(types.AuctionCollectorName)
+				resp := network.App.BankKeeper.GetBalance(network.GetContext(), auctionCollectorAddress, utils.BaseDenom)
+				assert.Equal(t, resp.Amount.Equal(sdk.ZeroInt()), true, "auction collector not empty")
+			},
+			input: func() *types.MsgDepositCoin {
+				return &types.MsgDepositCoin{
+					Sender: validSenderKey.AccAddr.String(),
+					Amount: sdk.NewCoin(utils.BaseDenom, depositAmount),
+				}
+			},
+			postCheck: func() {
+				auctionCollectorAddress := network.App.AccountKeeper.GetModuleAddress(types.AuctionCollectorName)
+				resp := network.App.BankKeeper.GetBalance(network.GetContext(), auctionCollectorAddress, utils.BaseDenom)
+				assert.Equal(t, resp.Amount, depositAmount, "expected the auction collector to have the deposit")
+			},
```

Comment:
having this postcheck function is unnecessarily complex IMO because it's only used once - it would be clearer to just have a `expectedDepositAmount` for the testcases and then check the deposited amount in the success case `if !expErr`
```suggestion
			expDepositAmount: depositAmount
```

since only the var is used, you don't even need to add it to the test cases technically, although it might make sense if there is a test case added some time in the future